Lib/test/test_tabnanny.py

```diff
@@ -0,0 +1,344 @@
+"""Testing `tabnanny` module.
+
+Glossary:
+    * errored    : Whitespace related problems present in file.
+"""
+from unittest import TestCase, mock
+from unittest import mock
+import tabnanny
+import tokenize
+import tempfile
+import textwrap
+from test.support import (captured_stderr, captured_stdout, script_helper,
+                          findfile, unlink)
+
+
+SOURCE_CODES = {
+    "incomplete_expression": (
+        'fruits = [\n'
+        '    "Apple",\n'
+        '    "Orange",\n'
+        '    "Banana",\n'
+        '\n'
+        'print(fruits)\n'
+    ),
+    "wrong_indented": (
+        'if True:\n'
+        '    print("hello")\n'
+        '  print("world")\n'
+        'else:\n'
+        '    print("else called")\n'
+    ),
+    "nannynag_errored": (
+        'if True:\n'
+        ' \tprint("hello")\n'
+        '\tprint("world")\n'
+        'else:\n'
+        '    print("else called")\n'
+    ),
+    "error_free": (
+        'if True:\n'
+        '    print("hello")\n'
+        '    print("world")\n'
+        'else:\n'
+        '    print("else called")\n'
+    ),
+    "tab_space_errored_1": (
+        'def my_func():\n'
+        '\t  print("hello world")\n'
+        '\t  if True:\n'
+        '\t\tprint("If called")'
+    ),
+    "tab_space_errored_2": (
+        'def my_func():\n'
+        '\t\tprint("Hello world")\n'
+        '\t\tif True:\n'
+        '\t        print("If called")'
+    )
+}
+
+
+class TemporaryPyFile:
+    """Create a temporary python source code file."""
+
+    def __init__(self, source_code='', directory=None):
+        self.source_code = source_code
+        self.dir = directory
+
+    def __enter__(self):
+        with tempfile.NamedTemporaryFile(
+            mode='w', dir=self.dir, suffix=".py", delete=False
+        ) as f:
+            f.write(self.source_code)
+        self.file_path = f.name
+        return self.file_path
+
+    def __exit__(self, exc_type, exc_value, exc_traceback):
+        unlink(self.file_path)
+        return exc_type is None
```

Comment:
I'm not sure that returning True is correct here. I just suggest to remove the "return" line.

Reply:
I agree, returning True isn't making sense here. Thanks for commenting.